hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1395,6 +1399,11 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 
 	// Reconcile the Ignition server
 	if !controlplaneOperatorManagesIgnitionServer {
+		ownerRef := config.OwnerRefFrom(hcp)
+		if r.ReleaseVersion.Major == 4 && r.ReleaseVersion.Minor < 12 {
+			ownerRef.Reference.APIVersion = hyperv1alpha1.GroupVersion.String()
+		}
```

Comment:
nit: put this logic in a single function that can be referenced called from wherever it's needed

Reply:
Extracted to common lib and added unit test